ui/admin/app/routes/scopes/scope/sessions/index.js

```diff
@@ -4,5 +4,200 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
+import runEvery from 'ember-pollster/decorators/route/run-every';
+import config from '../../../../config/environment';
 
-export default class ScopesScopeSessionsIndexRoute extends Route {}
+const POLL_TIMEOUT_SECONDS = config.sessionPollingTimeoutSeconds;
+
+export default class ScopesScopeSessionsIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service resourceFilterStore;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    users: {
+      refreshModel: true,
+      replace: true,
+    },
+    targets: {
+      refreshModel: true,
+      replace: true,
+    },
+    status: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  allUsers;
+  allSessions;
+
+  @runEvery(POLL_TIMEOUT_SECONDS * 1000)
+  poller() {
+    return this.refresh();
+  }
+
+  /**
+   * Loads all sessions under the current scope and encapsulates them into
+   * an array of objects containing their associated users and targets.
+   * @return {Promise{[{sessions: [SessionModel], allSessions: [SessionModel], allUsers: [UserModel], allTargets: [TargetModel], totalItems: number}]}}
+   */
+  async model({ search, users, targets, status, page, pageSize }) {
+    const { id: scope_id } = this.modelFor('scopes.scope');
+    const filters = {
+      status: [],
+      user_id: [],
+      target_id: [],
+    };
+    users.forEach((user) => {
+      filters.user_id.push({ equals: user });
+    });
+    targets.forEach((target) => {
+      filters.target_id.push({ equals: target });
+    });
+    status.forEach((item) => {
+      filters.status.push({ equals: item });
+    });
+
+    const queryOptions = {
+      scope_id,
+      recursive: true,
+      query: { search, filters },
+      page,
+      pageSize,
+      force_refresh: true,
+    };
+
+    const sessions = await this.store.query('session', queryOptions);
+    const totalItems = sessions.meta?.totalItems;
+
+    sessions.forEach(async (session) => {
+      if (session.user_id) {
+        session.user = await this.store.findRecord('user', session.user_id);
+      }
+      if (session.target_id) {
+        await this.store.findRecord('target', session.target_id);
+      }
+    });
+
+    // Query all sessions, users, and targets for defining filtering values if entering route for the first time
```

Comment:
I think this causes issues with displaying filters because even after polling refreshes the page, not all targets are displayed in the dropdown. The route is not being entered for the first time, so I'm assuming this gets skipped. 

Another thing, I think that retrieving allSessions might not be necessary since it's not needed for the filters.